lang/LangPrimSource/PyrPrimitive.cpp

```diff
@@ -390,6 +390,10 @@ int prFloat_AsStringPrec(struct VMGlobals *g, int numArgsPushed)
 	if (err) return err;
 
 	char fmt[8], str[256];
+	// if our precision is bigger than our stringsize, we can generate a very nasty buffer overflow here. So
+	if( precision <= 0 ) precision = 1;
+	if( precision >= 200 ) precision = 200; // Nothing is that big anyway. And we know we will be smaller than our 256 char string
```

Comment:
Welcome! Looking at my commit my comment is a bit imprecise and misleading when i say "nothing is that big anyway" when what I meant is "nothing's precision is that big anyway" - so thanks for reviewing and asking!